File: org.opencompare/play-app/app/controllers/Application.java
package controllers;

import model.Database;
import model.DatabasePCM;
import model.PCMInfo;
import play.libs.Json;
import play.mvc.Controller;
import play.mvc.Result;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.List;

@Singleton
public class Application extends Controller {


    public Result index() {
        return ok(views.html.index.render());
    }

    public Result aboutProject() {
        return ok(views.html.aboutProject.render());
    }

    public Result aboutPrivacyPolicy() {
        return ok(views.html.aboutPrivacyPolicy.render());
    }


    public Result list(int limit, int page) {
        List<PCMInfo> pcms = Database.INSTANCE.list(limit, page);
        int count = (int) Database.INSTANCE.count();
        int nbPages = count / limit;
        if (count % limit != 0) {
            nbPages++;
        }
        return ok(views.html.list.render(pcms, limit, page, nbPages));
    }

    public Result search(String request) {

        // TODO : find PCMs named "request" or with a product named "request"
        List<PCMInfo> results = Database.INSTANCE.search(request);

        return ok(views.html.search.render(request, results));
    }


    public Result edit(String id) {
        boolean exists = Database.INSTANCE.exists(id);
        if (exists) {
            return ok(views.html.edit.render(id, null, null));
        } else {
            return ok(views.html.edit.render(null, null, null));
        }

    }

    public Result create() {
        return ok(views.html.create.render(null, null, null));
    }

    public Result importer(String ext) {
        if (ext.equals("csv")) {
            return ok(views.html.edit.render(null, null, "CsvImport"));
        } else if (ext.equals("html")) {
            return ok(views.html.edit.render(null, null, "HtmlImport"));
        } else if (ext.equals("wikipedia")) {
            return ok(views.html.edit.render(null, null, "WikipediaImport"));
        }
        return notFound();
    }

    public Result embedPCM(String id) {
        boolean exists = Database.INSTANCE.exists(id);
        if (exists) {
            return ok(views.html.embed.render(id, null, null));
        } else {
            return ok(views.html.embed.render(null, null, null));
        }

    }

    public Result embed() {
        return ok(views.html.embed.render(null, null, null));
    }

}

File: org.opencompare/play-app/app/controllers/Application.scala
package controllers

import javax.inject._

import com.mohiva.play.silhouette.api.{Environment, Silhouette}
import com.mohiva.play.silhouette.impl.authenticators.CookieAuthenticator
import models.{Database, User}
import play.api.i18n.{I18nSupport, MessagesApi}
import play.api.mvc.{Action, Controller}

@Singleton
class Application @Inject() (val messagesApi: MessagesApi, val env: Environment[User, CookieAuthenticator]) extends BaseController {

    def index = UserAwareAction { implicit request =>
      Ok(views.html.index(Database.getLastHTMLSources(5)))
    }

    def aboutProject = UserAwareAction { implicit request =>
        Ok(views.html.aboutProject())
    }

    def aboutPrivacyPolicy = UserAwareAction { implicit request =>
        Ok(views.html.aboutPrivacyPolicy())
    }


    def list(limit : Int, page : Int) = UserAwareAction { implicit request =>
        val pcms = Database.list(limit, page).toList
        val count = Database.count().toInt
        var nbPages = count / limit
        if (count % limit != 0) {
            nbPages = nbPages + 1
        }
        Ok(views.html.list(pcms, limit, page, nbPages))
    }

    def search(searchedString : String) = UserAwareAction { implicit request =>

        // TODO : find PCMs named "request" or with a product named "request"
        val results = Database.search(searchedString).toList

        Ok(views.html.search(searchedString, results))
    }


    def edit(id : String) = UserAwareAction { implicit request =>
        val exists = Database.exists(id)
        if (exists) {
            Ok(views.html.edit(id, null, null))
        } else {
            Ok(views.html.edit(null, null, null))
        }

    }

    def create = UserAwareAction { implicit request =>
        Ok(views.html.edit(null, null, null, true))
    }

    def importer(ext : String) = UserAwareAction { implicit request =>
        ext match {
            case "csv" => Ok(views.html.edit(null, null, "Csv"))
            case "html" => Ok(views.html.edit(null, null, "Html"))
            case "wikipedia" => Ok(views.html.edit(null, null, "MediaWiki"))
            case _ => NotFound
        }
    }

    def embed(id : String) = UserAwareAction { implicit request =>
        val exists = Database.exists(id)
        if (exists) {
            Ok(views.html.embed(id, null, null))
        } else {
            Ok(views.html.embed(null, null, null))
        }

    }

}
